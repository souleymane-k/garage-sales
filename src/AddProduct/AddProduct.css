*{
  box-sizing: border-box;
  /* background-image: url(../../souleymane-3.jpg); */
}

fieldset {
    max-width: 60%;
    margin: 30px auto;
    height:640px;
    /* display: flex; */
    border: none;
    border-radius: 4px;
    box-shadow: -1px 1px 5px 4px black; 
    padding: 3px;
    background-color: #EAF6F6;
  }
  h2{
    color:#551A8B;
  }
  .form-group{
    margin: 20px 20px  20px 20px;
  
  }
  .form-group-select{
    margin: 20px 20px  20px 20px;
    width:95%;
    height:50px;
  }
  /* .add-result{
    margin:100px; */
    /* height:1000px; */
  /* } */
  label{
    color:#551A8B;
  }
  .form__button{
    /* position:absolute; */
    margin: 0 auto;
    height:35px;
    
} 
.form__button_cancel{
  /* margin: 10px 10px; */
  height:35px;


}
.form__button_cancel:hover{
  background-color:red;
  color:white;

}

.form__button:hover{
  background-color:green;
  color:white;

}
.add-product{
  /* background-image: url(../../souleymane-3.jpg); */
  height:1000px;
}
.form__button__group{
  margin: 20px 300px;
}
@media all and (max-width: 640px) {

  .form-group-select{
    margin: 20px 20px  20px 20px;
    width:86%;
    height:50px;
  }
  .form__button__group{
    margin: 20px 70px;
  }
  }