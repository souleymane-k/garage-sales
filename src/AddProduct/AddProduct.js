import React, { Component } from 'react';
import ApiContext from '../contexts/ApiContext';
import TokenService from '../services/TokenService'
import './AddProduct.css';
import config from '../config.js'

class AddProduct extends Component {
  constructor(props) {
    super(props);
    this.state = {
      product_name: {
        value: " ",
        touched: false
      },
      product_price: {
        value: " ",
        touched: false
      },
      date_posted: {
        value: " ",
        touched: false
      },
      description: {
        value: " ",
        touched: false
      },
      
      error: null,
    };
  }

  static contextType = ApiContext;

  updateProduct_name = (product_name) => {
    this.setState({ product_name: { value: product_name, touched: true } });
  }
  updateProduct_price = (product_price) => {
    this.setState({ product_price: { value: product_price, touched: true } });
  }
  updateDate_posted = (date_posted) => {
    this.setState({ date_posted: { value: date_posted, touched: true } });
  }
  updateDescription = (description) => {
    this.setState({ description: { value: description, touched: true } });
  }
  
  handleSubmit = e => {
    e.preventDefault();
    const {product_name, product_price,date_posted,description } = e.target;
    const product = {
      product_name: product_name.value,
      product_price: product_price.value,
      date_posted: date_posted.value,
      description: description.value,
    }
    console.log('message')

    this.setState({ error: null })
    console.log(`${config.API_ENDPOINT}/products`, {
      method: 'POST',
      body: JSON.stringify(product),
      headers: {
        'content-type': 'application/json',
        'Accept': 'application/json',
        'authorization': `bearer ${TokenService.getAuthToken()}`
      }
    })
    fetch(`${config.API_ENDPOINT}/products`, {
      method: 'POST',
      body: JSON.stringify(product),
      headers: {
        'content-type': 'application/json',
        'Accept': 'application/json',
        'authorization': `bearer ${TokenService.getAuthToken()}`
      }
    })
      .then(res => {
        if (!res.ok) {
          return res.json().then(error => {
            throw error
          })
        }
        return res.json()
      })

      .then(data => {
        product_name.value = '';
        product_price.value = '';
        date_posted.value = '';
        description.value = '';
        this.props.history.push('/home')
      })
      .catch(error => {
        console.log(error)
        this.setState({ error })
      })
  }


  render() {
    return (
      <form className="add-product" onSubmit={e => this.handleSubmit(e)}>
        <fieldset>
          <div>
            <h2>New Product</h2>
          </div>
          <div className='form-group'>
            <label htmlFor="product_name">Product*</label>
            <input
              type="text"
              className="form__input"
              name="product_name"
              id="product_name"
              required
              onChange={e => this.updateProduct_name(e.target.value)}
            />
          </div>
          <div className='form-group'>
            <label htmlFor="product_price">Price*</label>
            <input
              type="integer"
              className="form__input"
              name="product_price"
              id="product_price"
              required
              onChange={e => this.updateProduct_price(e.target.value)}
            />
          </div>
          <div className='form-group'>
            <label htmlFor="date_posted">date*</label>
            <input
              type="integer"
              className="form__input"
              name="date_posted"
              id="date_posted"
              required
              onChange={e => this.updateDate_posted(e.target.value)}
            />
          </div>
          <div className='form-group'>
            <label htmlFor="description">Description *</label>
            <input
              type="text"
              className="form__input"
              name="description"
              id="description"
              required
              onChange={e => this.updateDescription(e.target.value)}
            />
          </div>
        
          <div className="form__button__group">
            <button
              type="submit"
              className="form__button"
            // onClick={this.handleSubmit}
            >

              Save Product
            </button>


            <button type="reset" className="form__button_cancel"
              onClick={this.handleClickCancel}
            >
              Cancel
            </button>
          </div>

        </fieldset>
      </form>

    )
  }
}

export default AddProduct;

