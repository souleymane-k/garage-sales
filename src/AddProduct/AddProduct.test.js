import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddProduct from './Addproduct';

describe(`AddProduct  component`, () => {
  it('renders the complete form', () => {
    const wrapper = shallow(<AddProduct  />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AddProduct  />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
})
