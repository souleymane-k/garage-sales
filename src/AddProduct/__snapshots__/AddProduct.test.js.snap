// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddProduct  component renders the complete form 1`] = `
<form
  className="add-product"
  onSubmit={[Function]}
>
  <fieldset>
    <div>
      <h2>
        New Product
      </h2>
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="product_name"
      >
        Product*
      </label>
      <input
        className="form__input"
        id="product_name"
        name="product_name"
        onChange={[Function]}
        required={true}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="product_price"
      >
        Price*
      </label>
      <input
        className="form__input"
        id="product_price"
        name="product_price"
        onChange={[Function]}
        required={true}
        type="integer"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="date_posted"
      >
        date*
      </label>
      <input
        className="form__input"
        id="date_posted"
        name="date_posted"
        onChange={[Function]}
        required={true}
        type="integer"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="description"
      >
        Description *
      </label>
      <input
        className="form__input"
        id="description"
        name="description"
        onChange={[Function]}
        required={true}
        type="text"
      />
    </div>
    <div
      className="form__button__group"
    >
      <button
        className="form__button"
        type="submit"
      >
        Save Product
      </button>
      <button
        className="form__button_cancel"
        type="reset"
      >
        Cancel
      </button>
    </div>
  </fieldset>
</form>
`;
